webapp/src/js/reducers/reports.js

```diff
@@ -1,13 +1,54 @@
+const _ = require('underscore');
 const actionTypes = require('../actions/actionTypes');
 const initialState = {
+  selected: []
 };
 
 module.exports = function(state, action) {
   if (typeof state === 'undefined') {
     state = initialState;
   }
 
+  let filteredSelected;
+  let selected;
```

Comment:
Is it wise to persist state here between actions? Seems preferable to have multiple scoped (const) variables with the same name per case block rather than one with a broad scope that persists between actions. Or maybe I'm missing something?

Reply:
This function is actually called every time an action is dispatched, thus these two variables are re-declared on every action and so no persisted state 🙂